CondCore/CondHDF5ESSource/plugins/HDF5ProductResolver.cc

```diff
@@ -0,0 +1,249 @@
+// -*- C++ -*-
+//
+// Package:     CondCore/HDF5ESSource
+// Class  :     HDF5ProductResolver
+//
+// Implementation:
+//     [Notes on implementation]
+//
+// Original Author:  Christopher Jones
+//         Created:  Tue, 20 Jun 2023 13:52:59 GMT
+//
+
+// system include files
+#include <iostream>
+#include <fstream>
+#include <cassert>
+#include "zlib.h"
+#include "lzma.h"
+
+// user include files
+#include "HDF5ProductResolver.h"
+#include "convertSyncValue.h"
+#include "FWCore/Framework/interface/EventSetupRecordImpl.h"
+#include "FWCore/ServiceRegistry/interface/ESModuleCallingContext.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+#include "FWCore/Concurrency/interface/SerialTaskQueue.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "h5_DataSet.h"
+#include "h5_Attribute.h"
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+HDF5ProductResolver::HDF5ProductResolver(edm::SerialTaskQueue* iQueue,
+                                         std::unique_ptr<cond::serialization::SerializationHelperBase> iHelper,
+                                         cms::h5::File const* iFile,
+                                         std::string const& iFileName,
+                                         cond::hdf5::Compression iCompression,
+                                         cond::hdf5::Record const* iRecord,
+                                         cond::hdf5::DataProduct const* iDataProduct)
+    : edm::eventsetup::ESSourceProductResolverBase(),
+      queue_(iQueue),
+      helper_(std::move(iHelper)),
+      file_(iFile),
+      fileName_(iFileName),
+      record_(iRecord),
+      dataProduct_(iDataProduct),
+      compression_(iCompression) {}
+
+// HDF5ProductResolver::HDF5ProductResolver(const HDF5ProductResolver& rhs)
+// {
+//    // do actual copying here;
+// }
+
+HDF5ProductResolver::~HDF5ProductResolver() {}
+
+//
+// member functions
+//
+
+void HDF5ProductResolver::prefetchAsyncImpl(edm::WaitingTaskHolder iTask,
+                                            edm::eventsetup::EventSetupRecordImpl const& iRecord,
+                                            edm::eventsetup::DataKey const& iKey,
+                                            edm::EventSetupImpl const*,
+                                            edm::ServiceToken const&,
+                                            edm::ESParentContext const& iParent) {
+  prefetchAsyncImplTemplate(
+      [this, iov = iRecord.validityInterval(), iParent, &iRecord](auto& iGroup, auto iActivity) {
+        queue_->push(iGroup, [this, &iGroup, act = std::move(iActivity), iov, iParent, &iRecord] {
+          try {
```

Comment:
How about
```suggestion
          CMS_SA_ALLOW try {
```
? (won't help with the "CMSSW code rule checker", but at least silences the static analyzer warning)

Reply:
done